src/proto/factories/callback_factory.cpp

```diff
@@ -97,14 +97,11 @@ lbann_callback* construct_callback(lbann_comm* comm,
   //////////////////////////////////////////////////////////////
 
   if (proto_cb.has_ltfb()) {
-    auto&& m = parse_list<>(proto_cb.ltfb().eval_metrics());
-    auto&& w = parse_list<>(proto_cb.ltfb().weights_tosend());
-    std::unordered_set<std::string> metric_names(m.begin(), m.end());
-    std::unordered_set<std::string> weight_names(w.begin(), w.end());
-    return new lbann_callback_ltfb(proto_cb.ltfb().round_size(),
-                                   metric_names,
-                                   proto_cb.ltfb().increasing_metric_mode(),
-                                   weight_names,
+    const auto& params = proto_cb.ltfb();
+    return new lbann_callback_ltfb(params.batch_interval(),
+                                   params.metric(),
```

Comment:
We are not using multiple metrics now but how would this support space-delimited metric (e.g., categorical accuracy) and multiple metrics with the same name as metric specified one of the metrics for training/inference ?

Reply:
- I've removed support for multiple metrics. At the end of the day, LTFB will choose winners based on a single number. If it's the average of multiple metrics, the user can compute the mean manually with a weighted sum layer. Similar tricks can be used for the min/max of multiple layers, although there will be some mathematical differences (average of maxes rather than max of averages).
- Since we only expect one metric, there's no problem if there are spaces in the metric name. Come to think of it, though, we may want to force metric names to have no spaces (like in layer names).
- I think we should require the metric name to be a unique identifier, like the layer name. If we allow metric names to be duplicated, we'll need to define a "metric ID" or something similar to be a unique identifier, so the practitioner won't save any effort.